### PR TITLE
qalculate-qt: init at 4.2.0

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/math/qalculate-qt/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "qalculate";
+    repo = "qalculate-qt";
+    rev = "v${version}";
+    sha256 = "sha256-7H1nQLJBiuTj/GwojfOPpRbDseOHvLa94LK+bXvLhws=";
+  };
+
+  nativeBuildInputs = [ qmake intltool pkg-config wrapQtAppsHook ];
+  buildInputs = [ libqalculate qtbase qttools ];
+  enableParallelBuilding = true;
```
Comment (romildo) [inline pkgs/applications/science/math/qalculate-qt/default.nix:16]:
```suggestion
```

`enableParallelBuilding` already defaults to true with `qmake`.

File: pkgs/applications/science/math/qalculate-qt/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description = "The ultimate desktop calculator";
+    homepage = "http://qalculate.github.io";
+    maintainers = with maintainers; [ _4825764518 ];
+    license = licenses.gpl2Plus;
+    platforms = [ "x86_64-linux" ];
```
Comment (romildo) [inline pkgs/applications/science/math/qalculate-qt/default.nix:23]: Maybe the platform can be more general, like `unix` or `linux`.

**Review by romildo — CHANGES_REQUESTED** [review]

**Comment by romildo** [comment]
Result of `nixpkgs-review pr 185606` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qalculate-qt</li>
  </ul>
</details>

File: pkgs/applications/science/math/qalculate-qt/default.nix (hunk 2 above)
Comment (ajs124) [reply pkgs/applications/science/math/qalculate-qt/default.nix]: Doesn't seem like it. It fails to build on aarch64-linux for me and ofborg. So let's go back to `x86_64-linux` and get this merged.

**Review by romildo — APPROVED** [review]

**Pull request merged**